src/gui/foldercreationdialog.cpp

```diff
@@ -0,0 +1,80 @@
+#include "foldercreationdialog.h"
+#include "ui_foldercreationdialog.h"
+
+#include <limits>
+
+#include <QDir>
+#include <QMessageBox>
+
+namespace  {
+  const QString SuggestedFolderNamePrefix = QObject::tr("New folder");
+};
+
+FolderCreationDialog::FolderCreationDialog(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::FolderCreationDialog)
+{
+    ui->setupUi(this);
+
+    ui->labelErrorMessage->setVisible(false);
+
+    setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+}
+
+FolderCreationDialog::~FolderCreationDialog()
+{
+    delete ui;
+}
+
+void FolderCreationDialog::setDestination(const QString& destination)
+{
+    Q_ASSERT(!destination.endsWith('/'));
+    _destination = destination;
+}
+
+void FolderCreationDialog::showEvent(QShowEvent *event)
+{
+    const auto newFolderFullPath = _destination + "/" + SuggestedFolderNamePrefix;
```

Comment:
Wouldn't it make more sense to have that inside `setDestination` instead?

I mean we could always construct cases where the dialog is displayed first and setDestination called a bit later asynchronously.

Reply:
@er-vin My intent was to make sure the dialog is always initialized when it is shown. But, I haven't made this intent clear. I will change the constructor and remove the setter now.

Reply:
Note I didn't mind the setter. But OK let's go with a ctor parameter instead. You won't need show event anymore though, this can all be done in the ctor then.

Reply:
@er-vin Alrighty then. Removed.